openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -289,6 +290,11 @@ def _update_extended_profile_if_needed(data, user_profile):
         user_profile.set_meta(meta)
         user_profile.save()
 
+def _update_state_if_needed(data, user_profile):
+    # If the country was changed to something other than US, remove the state.
+    if "country" in data and data['country'] != 'US':
```

Comment:
You can squish `"country" in data and data['country']` into `data.get("country")`. `d.get(x)` will either return `d["x"]` if it exists, or None.

Reply:
Also discussed via Slack. `if data.get("country") != 'US':` won’t quite work here because of the way fields get patched one at a time on the account page.

(This would make changes to the state field itself—which come in as `{state: "MA"}`, for example, where `data.get("country")` would be `None` and therefore `!= 'US'`—execute the code that wipes out state! We don‘t want that.)